GitCommands/RevisionReader.cs

```diff
@@ -98,6 +98,7 @@ public sealed class RevisionReader : IDisposable
                 /* Author date     */ "%at%n" +
                 /* Commit date     */ "%ct%n" +
                 /* Encoding        */ "%e%n" +
+                /* Abbrev Object ID*/ "%h%n" +
```

Comment:
Should we fetch the abbrev id or shorten the object id (approx length as used elsewhere in GE)
This is reversing (a very small fraction) of what @drewnoakes improved

Reply:
We shorten the objectid elsewhere because git doesn't provide it I imagine (or because it will need to do an extra git call to get it).
Here git is providing them for us.
I'm for letting git do what it is good for (it's provided for free) and don't try to reimplement its stuff by adding code that is harder to maintain...

For my information, how do we get the length of the abbreviated hash elsewhere in the application?

Reply:
> For my information, how do we get the length of the abbreviated hash elsewhere in the application?

Hardcoded to 10 characters or so. This will fail for really large repos like Linux kernel.

Abbrev adds to data that need to be stored, something that drew just trimmed to get down loading times at lot (about half). Some of that was due to less stored data, this adds some again. Not much. The new ObjectId could be a more efficient way to store the data. See #4988 for more info.
I would probably like to see this anyway.

Reply:
@drewnoakes Comments on data retrieved?

Reply:
We definitely want the full hash in memory.

As for whether we bring back the abbreviated form as well, I'm on the fence. I experimented in another branch (PR coming soon) with just showing enough characters to fill whatever width the user gives to the column without ellipsis. I felt that was pretty nice.

I would avoid reading the string. We don't need to allocate it in memory, as it's redundant given we have the full ID in another field. All we need is `public byte UnambiguousObjectIdLength { get; }` on the revision class. We can then use `ObjectId.ToShortString` as needed.

Asking git to compute the length makes the log command take about 20% longer on my machine (tested on the command line). Not the end of the world, but not free. Again, I'm on the fence.

Run the test yourself with:

```bash
$ time git log -z --pretty=format:"%H%T%P%n%at%n%ct%n%e%n%aN%n%aE%n%cN%n%cE%n%s%n%n%b" --topo-order --all --boundary --not --glob=notes --not  --max-count="100000" -- > /dev/null

$ time git log -z --pretty=format:"%H%T%P%n%at%n%ct%n%e%n%h%n%aN%n%aE%n%cN%n%cE%n%s%n%n%b" --topo-order --all --boundary --not --glob=notes --not  --max-count="100000" -- > /dev/null
```

We can always compute the unambiguous length on demand when viewing single commits.

Reply:
I lean on to not getting the short sha automatically but to estimate it like elsewhere in GE right now.
"Optimize now, make complete later if needed" is reverse what you normally do, except that we know this handling need to be optimized.

Reply:
```
$ time git log -z --pretty=format:"%H%T%P%n%at%n%ct%n%e%n%aN%n%aE%n%cN%n%cE%n%s%n%n%b" --topo-order --all --boundary --
not --glob=notes --not  --max-count="100000" -- > /dev/null

real    0m0.310s
user    0m0.015s
sys     0m0.000s

$ time git log -z --pretty=format:"%H%T%P%n%at%n%ct%n%e%n%h%n%aN%n%aE%n%cN%n%cE%n%s%n%n%b" --topo-order --all --boundar
y --not --glob=notes --not  --max-count="100000" -- > /dev/null

real    0m0.723s
user    0m0.000s
sys     0m0.016s
```

Reply:
Wow that's a big difference/price to pay.

Reply:
this is on W10 running under Parallels on a MBP, I consider this instance quite snappy compared to my work environment (I'm about a week away from running a test on it).

Reply:
Here's my work laptop against GE repo

```
$ time git log -z --pretty=format:"%H%T%P%n%at%n%ct%n%e%n%aN%n%aE%n%cN%n%cE%n%s%n%n%b" --topo-order --all --boundary --not --glob=notes --not  --max-count="100000" -- > /dev/null

real    0m1.491s
user    0m0.000s
sys     0m0.078s

$ time git log -z --pretty=format:"%H%T%P%n%at%n%ct%n%e%n%h%n%aN%n%aE%n%cN%n%cE%n%s%n%n%b" --topo-order --all --boundary --not --glob=notes --not  --max-count="100000" -- > /dev/null

real    0m6.588s
user    0m0.000s
sys     0m0.077s
```

Reply:
Oh wow. That's a huge difference. What spec does your work laptop have (CPU/RAM/HDD)?

Reply:
the actual specs aren't too bad, i7vpro, 16G with a ssd.
the problem is that OS is virtualised and hamstrung by mcafee and other corporate bloatware... it is a very frustrating devex.